rclcpp_components/src/component_manager.cpp

```diff
@@ -215,9 +227,17 @@ ComponentManager::on_load_node(
       }
 
       auto node = node_wrappers_[node_id].get_node_base_interface();
-      if (auto exec = executor_.lock()) {
-        exec->add_node(node, true);
+      if (use_dedicated_thread_) {
+        auto exec = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+        exec->add_node(node);
+        executors_[node_id] = exec;
+        executor_threads_[node_id] = std::make_shared<std::thread>([exec]() {exec->spin();});
+      } else {
+        if (auto exec = executor_.lock()) {
+          exec->add_node(node, true);
+        }
       }
+
```

Comment:
Is this new line is required to add? if not, we would probably want to avoid the unnecessary change.